# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/IOThreadChild.h
# Commit: 790fd8710dd7
# Full Hash: 790fd8710dd7bfa6085379e7f90b043e7aff383f
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 12a: Initialize NodeController when creating IO thread, r=handyman
#   
#   This also consumes the existing channel created when launching a process to
#   create the the conneciton required by NodeController for communicating between
#   processes. In part 12b, consumers of the broken APIs will be adjusted to use
# ==============================================================================

diff -r 7b7819250f79 -r 790fd8710dd7 ipc/glue/IOThreadChild.h
--- a/ipc/glue/IOThreadChild.h	Tue Jun 22 18:17:22 2021 +0000
+++ b/ipc/glue/IOThreadChild.h	Tue Jun 22 18:17:22 2021 +0000
@@ -8,6 +8,7 @@
 #define dom_plugins_IOThreadChild_h 1
 
 #include "chrome/common/child_thread.h"
+#include "mozilla/ipc/NodeController.h"
 
 namespace mozilla {
 namespace ipc {
@@ -28,8 +29,8 @@
     return IOThreadChild::current()->Thread::message_loop();
   }
 
-  static UniquePtr<IPC::Channel> TakeChannel() {
-    return IOThreadChild::current()->ChildThread::TakeChannel();
+  static ScopedPort TakeInitialPort() {
+    return IOThreadChild::current()->ChildThread::TakeInitialPort();
   }
 
  protected:
